### ISSUE TITLE
Heap Size too small

### ISSUE THREAD

**Issue by odelphi** [Audiveris/audiveris#409]
Currently the heap size in "build.grable" is 1g.  For a modern system, built in the last 4 years, 1g is way too small.  Should be at least 2g.  Could there be a new Windows build that increases the heap size to 2g?  I am too much of a noob to do it myself.  Even better would be to make heap size a user option rather than it being hard-coded.  Looks like it would be a fairly easy change.
Thanks.

**Comment by hbitteur** [comment]
If you launch Audiveris via gradlew, you could specify memory limits there. Have a look at the build.gradle file, in the "run" section you'll find minHeapSize and maxHeapSize parameters. 

If you don't want to edit this build.gradle file, there must be a way to override these values when you launch gradlew. I don't have the answer immediately, but a Google search on "gradle memory" gave me many pages, beginning by [this one](https://riptutorial.com/gradle/example/11911/tuning-jvm-memory-usage-parameters-for-gradle#:~:text=The%20default%20memory%20usage%20settings%20for%20Gradle%20builds,memory%20allocation%20for%20permanent%20%22internal%22%20objects%20of%20256MB.) which says to use gradle.properties to specify memory limits for a given project.

If you launch Audiveris by one of its scripts available in build/scripts folder (Audiveris and Audiveris.bat), you have additional possibilities there. Read these scripts, about DEFAULT_JVM_OPTS, JAVA_OPTS and AUDIVERIS_OPTS.

**Comment by odelphi** [comment]
Thanks.  But if I change build.gradle don't I have to recompile the program?  That file is not on my computer.  I downloaded the binary version of the program (.exe).  It doesn't put that file on your system. Your solutions only work if you recompile the program.  
 However, I have an update.  I bit the bullet and tried to recompile the program myself.  Had to download about a gig of sh*t (JDK, GIT and crap) but I did it, changed the build.gradle file (like you suggested) recompiled the program.  When I execute the program, I get a error about not being able to create a 2g heap. (you can change the maxheapsize to 2g).  I have 8g of RAM so I don't know.  Anywho, I put the heapsize back to 1g, recompiled and now it works.  A google search told me how to increase the Java Heap size.  I did that.  Maybe that did it because instead of having just 291M of memory available for the OCR, I have ~500M.  That has allowed me to scan at 600DPI, which helps the scan recognize 16th notes better.  I still get annoying bad rhythm errors.  I don't know what is causing it.  Musescore has a hard time reading MXL files that have errors, it causes all kinds of recognition problems.
I am having another odd problem.  Maybe you have advice.  When I read an MXL file into Musescore, it oddly won't read the last two staves.  Maybe it is a memory thing;  I don't know.

**Comment by hbitteur** [comment]
Let's keep focused on the topic at hand. If you have a non-related question, please file a new issue for that purpose. This will help the other users search and browse this forum.

This issue is supposed to deal with heap size. So let's get back to this topic only.

Could you be more precise in the way (or ways) you use to launch Audiveris software.  And then we'll see how we can modify Java heap size  in a way compatible with your launch mechanism.

**Comment by odelphi** [comment]
Sorry if I lost you.  Everything I said has to do with heap size/memory.  The last sentence is me wondering if low memory is causing the last two staves to be left off.  But continuing . . . .
As I stated, I recompiled audiveris which is giving me about 400MB more memory (don't know why).  I didn't do anything different except set a JAVA param, in the JAVA console, to increase heap size.  As I stated, when I increased maxheapsize in build.gradle and recompiled, I got a error that it can't create a 2g heap size so I went back to a 1g heap size.  
Currently,  I have to run the program via GIT, which is an annoyance. (Any help on this will be appreciated).  I have to run auditveris by executing (in a command line) "gradlaw.bat run".  Apparently, running Auditveris via GIT gives you more memory (?).
Thank you for helping.
